src/Api/Authorization/HeaderHandler.cs

```diff
@@ -89,7 +91,28 @@ protected override async Task HandleForbiddenAsync(AuthenticationProperties prop
                 Status = StatusCodes.Status403Forbidden,
                 Type = "https://docs.passwordless.dev/guide/errors.html#forbidden",
             },
-        });
+        };
+
+        var endpoint = Context.GetEndpoint();
+        if (endpoint == null)
+        {
+            await _problemDetailsService.WriteAsync(context);
+            return;
+        }
+
+        var policies = endpoint
+            .Metadata
+            .Where(x => x is AuthorizeAttribute)
+            .Cast<AuthorizeAttribute>()
+            .ToList();
+
+        if (policies.Count > 0)
+        {
+            context.ProblemDetails.Detail =
+                $"You are unable to view this resource because you do not have the required permissions. Required scopes: {string.Join(", ", policies.Select(x => x.Policy))}";
```

Comment:
Nit: maybe `access` is a better verb, considering it could be a `POST` request?

```suggestion
                $"You are unable to access this resource because you do not have the required permissions. Required scopes: {string.Join(", ", policies.Select(x => x.Policy))}";
```